htlcswitch/payment_result.go

```diff
@@ -115,9 +217,17 @@ func (store *networkResultStore) StoreResult(attemptID uint64,
 	store.attemptIDMtx.Lock(attemptID)
 	defer store.attemptIDMtx.Unlock(attemptID)
 
+	return store.storeResult(attemptID, result)
+}
+
+// storeResult is the internal version of StoreResult that assumes the caller
+// is holding the attemptIDMtx lock.
+func (store *networkResultStore) storeResult(attemptID uint64,
```

Comment:
I wonder if we can split this function to store and notification to subsribers, this would make it also more easy to follow the code, because we do not want to notify when we init an attempt.